drivers/input/input_touch.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2022 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/input/input_touch.h>
+
+void input_touchscreen_report_pos(const struct device *dev, uint32_t x, uint32_t y)
+{
+	const struct input_touchscreen_common_config *cfg = dev->config;
+
+	input_report_abs(dev,
+	cfg->swapped_x_y ? INPUT_ABS_Y : INPUT_ABS_X,
+	cfg->inverted_x ? cfg->screen_width - x : x,
+		false, K_FOREVER);
+	input_report_abs(dev,
+		cfg->swapped_x_y ? INPUT_ABS_X : INPUT_ABS_Y,
+		cfg->inverted_x ? cfg->screen_height - y : y,
+		false, K_FOREVER);
```

Comment:
I wouldn't be mad if you added some temporary variables here for sake of code redability

Reply:
Added temporary variables